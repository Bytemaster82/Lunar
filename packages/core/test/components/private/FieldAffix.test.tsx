import React from 'react';
import { shallow } from 'enzyme';
import FieldAffix from '../../../src/components/private/FieldAffix';

describe('<FieldAffix />', () => {
  it('renders before and after with different classes', () => {
    const wrapper = shallow(<FieldAffix before>Child</FieldAffix>).dive();
    const beforeClass = wrapper.prop('className');

    wrapper.setProps({
      before: false,
      after: true,
    });

    expect(wrapper.prop('className')).not.toBe(beforeClass);
  });

  it('renders compact', () => {
    const wrapper = shallow(
      <FieldAffix after compact>
        Child
      </FieldAffix>,
    ).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('errors if both props used', () => {
    expect(() => {
      shallow(
        <FieldAffix before after>
          Child
        </FieldAffix>,
      ).dive();
    }).toThrowError();
  });
});
