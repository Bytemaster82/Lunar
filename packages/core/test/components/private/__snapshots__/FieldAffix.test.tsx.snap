// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldAffix /> renders compact 1`] = `
<div
  className="affix_1ypg015-o_O-affix_compact_tue55j-o_O-affix_after_h7uxoa"
>
  Child
</div>
`;
